### PR TITLE
SOC-1473 | Convert controllers functionality to components and services

### PR DESCRIPTION
**Pull request by kvas-damian**
https://wikia-inc.atlassian.net/browse/SOC-1473

Discussions:
* move logic from controllers into Components and Services
* actions bubbling cleanup
* improved back links
* improved sorting

### CODE REVIEW

**Comment by rafalkalinski** [comment]
I'm sorry I can't find anything wrong.
LGTM

File: front/scripts/main/components/discussion-header.js [hunk 1]
```diff
@@ -6,8 +6,11 @@ export default Ember.Component.extend(
 		classNames: ['discussion-header', 'background-theme-color'],
 
 		discussionEditor: Ember.inject.service(),
+		discussionSort: Ember.inject.service(),
+
+		sortMessageKey: Ember.computed.oneWay('discussionSort.sortMessageKey'),
```
Comment (bkoval) [inline front/scripts/main/components/discussion-header.js]: Seems like this one is breaking alphabetical order :)

File: front/scripts/main/routes/discussion/forum.js [hunk 2]
```diff
@@ -16,22 +17,13 @@ export default DiscussionBaseRoute.extend(
 		 * @returns {Ember.RSVP.Promise}
 		 */
 		model(params) {
-			const sortBy = params.sortBy || this.defaultSortType;
+			if (params.sortBy) {
+				this.get('discussionSort').setSortBy(params.sortBy);
```
Comment (bkoval) [inline front/scripts/main/routes/discussion/forum.js]: Doesn't it need to fall back to default type anymore?

Comment (kvas-damian) [inline front/scripts/main/routes/discussion/forum.js]: Default was moved here: https://github.com/Wikia/mercury/pull/1825/files#diff-c53c5468df14eb37d4b65879cfe6559cR14

Comment (bkoval) [inline front/scripts/main/routes/discussion/forum.js]: :+1:

**Comment by bkoval** [comment]
LGTM

**Pull request merged**